frame/staking/src/offchain_election.rs

```diff
@@ -208,17 +375,38 @@ pub fn prepare_submission<T: Trait>(
 	let low_accuracy_assignment = sp_npos_elections::assignment_staked_to_ratio_normalized(staked)
 		.map_err(|e| OffchainElectionError::from(e))?;
 
-	// convert back to staked to compute the score in the receiver's accuracy. This can be done
-	// nicer, for now we do it as such since this code is not time-critical. This ensure that the
-	// score _predicted_ here is the same as the one computed on chain and you will not get a
-	// `PhragmenBogusScore` error. This is totally NOT needed if we don't do reduce. This whole
-	// _accuracy glitch_ happens because reduce breaks that assumption of rounding and **scale**.
-	// The initial phragmen results are computed in `OffchainAccuracy` and the initial `staked`
-	// assignment set is also all multiples of this value. After reduce, this no longer holds. Hence
-	// converting to ratio thereafter is not trivially reversible.
+	// compact encode the assignment.
+	let compact = CompactAssignments::from_assignment(
+		low_accuracy_assignment,
+		nominator_index,
+		validator_index,
+	)
+	.map_err(|e| OffchainElectionError::from(e))?;
+
+	// potentially reduce the size of the compact to fit weight.
+	let maximum_allowed_voters =
+		maximum_compact_len::<T::WeightInfo>(winners.len() as u32, size, maximum_weight);
+
+	crate::log!(info, "Maximum weight = {:?} // current weight = {:?} // maximum voters = {:?} // current votes = {:?}",
+		maximum_weight,
+		T::WeightInfo::submit_solution_better(
+				size.validators.into(),
+				size.nominators.into(),
+				compact.len() as u32,
+				winners.len() as u32,
+		),
+		maximum_allowed_voters,
+		compact.len(),
+	);
+
+	let compact = trim_to_weight::<T, _>(maximum_allowed_voters, compact, &nominator_index)?;
+
+	// re-compute the score. We re-create what the chain will do.
```

Comment:
The previous comment seemed much more detailed, is it outdated now?

Reply:
Now but it was more like a comment for me and by now I have learned that by heart.. but better to actually revert it. Fair point.

Reply:
or revert to something a bit shorter.